packages/block-editor/src/utils/use-can-block-toolbar-be-focused.js

```diff
@@ -9,40 +9,47 @@ import { isUnmodifiedDefaultBlock } from '@wordpress/blocks';
  */
 import { store as blockEditorStore } from '../store';
 import { unlock } from '../lock-unlock';
+import { useHasAnyBlockControls } from '../components/block-controls/use-has-block-controls';
 
 /**
  * Returns true if the block toolbar should be able to receive focus.
  *
  * @return {boolean} Whether the block toolbar should be able to receive focus
  */
 export function useCanBlockToolbarBeFocused() {
-	return useSelect( ( select ) => {
-		const {
-			__unstableGetEditorMode,
-			getBlock,
-			getSettings,
-			getSelectedBlockClientId,
-			getFirstMultiSelectedBlockClientId,
-		} = unlock( select( blockEditorStore ) );
+	const hasBlockControls = useHasAnyBlockControls();
```

Comment:
Use the Settings Icon and select "Hide Whitespace" and apply and reload. The only lines in this PR that changed are the ones related to `hasBlockControls`
<img width="1021" alt="Screenshot of this page with the hide whitespace option selected" src="https://github.com/WordPress/gutenberg/assets/967608/7d189cfb-da22-4134-a1cf-3c037ebd9d34">